### ISSUE TITLE
ICANN CZDS API

### ISSUE THREAD

**Issue by elliotwutingfeng** [trickest/zip#2]
A much easier approach is to grab the official `.zip` zonefile from [ICANN Centralized Zone Data Service](https://czds.icann.org) API.

Drawback being that it is unclear if publishing the domain names here is allowed by their Terms of Service.

Your batch processing workflow would be more useful for guessing domains with country-code TLDs like `.uk`, `.my` whose zonefiles are usually not available to the public.

**Comment by mhmdiaa** [comment]
Hi, great suggestion! We've applied for access to the TLD's zone file and when/if the request gets approved, we'll update the workflow to use that instead of the current approach. Thanks a lot!

**Comment by mhmdiaa** [comment]
The TLD zone file access has been approved and it's been integrated into the workflow. While it did add some new domains, I noticed that our initial wordlist approach also had some unique findings of its own that weren't in the zone file; there might be a delay between when a domain is registered and when it's added to the published zone file.

So we have decided to continue using our existing approach while incorporating the zone file as a complementary data source to improve the wordlist.

Thank you!
